src/core_modules/capture-core/components/Pages/Search/SearchPage.component.js

```diff
@@ -28,24 +41,52 @@ const getStyles = (theme: Theme) => ({
     },
     searchRow: {
         display: 'flex',
-        padding: '8px 0',
         alignItems: 'center',
         justifyContent: 'center',
     },
     searchRowTitle: {
         flexBasis: 200,
-        marginLeft: 16,
+        marginLeft: 8,
     },
     searchRowSelectElement: {
         width: '100%',
-        marginRight: 8,
+    },
+    searchButtonContainer: {
+        padding: theme.typography.pxToRem(10),
+        display: 'flex',
+        alignItems: 'center',
     },
 });
 
 
-const Index = ({ classes, trackedEntityTypesWithCorrelatedPrograms, preselectedProgram }: Props) => {
+const Index = ({
+    classes,
+    trackedEntityTypesWithCorrelatedPrograms,
+    preselectedProgram,
+    availableSearchOptions,
+    addFormIdToReduxStore,
+    forms,
+}: Props) => {
     const [selectedOption, setSelected] = useState(preselectedProgram);
 
+    // dan abramov suggest to stringify https://twitter.com/dan_abramov/status/1104414469629898754?lang=en
+    const stringifyPrograms = JSON.stringify(availableSearchOptions);
```

Comment:
Ties to a comment from the previous PR regarding memoization of objects from the container (and also one from this PR): If `reselect` is used for `availableSearchOptions` this isn't necessary. As far as I can understand the contents of `availableSearchOptions` will never change and it would be beneficial to keep the same object reference to avoid this 'workaround' (as well as doing the calculation for `availableSearchOptions` once).

Reply:
This will be reoccurring in all the current PRs since I havent used `reselect` Just `useMemo`. But is valid point. Can we make a 'refactoring' as a last PRs where we deal with the `reselect` and memoization?

Reply:
I am creating a new branch where this is gonna be included. 

However there is a little thing we need to puzzle out. `useEffect` will thrown a warning if we dont add each dependency it is used inside its callback. We are using the `availableSearchOptions` in the callback to add the `formId` in the redux store and therefore `useEffect` wont allow us to skip adding it in the array even if the object is memoized.

Silencing the linter is not recommended by the react team and therefore they suggest this workaround. This is why I ended up with it. I know it is not that intuitive but I am happy to find a better solution with your help

Reply:
You will add the `availableSearchOptions` as a dependency to `useEffect`. `availableSearchOptions` will never change if it is memoized, so the effect will never retrigger because of `availableSearchOptions`.

Reply:
I see. Would give it a try :)